app/controllers/rentals_controller.rb

```diff
@@ -11,9 +11,22 @@ def show; end
   def create
     @rental = current_user.rentals.new(rental_params)
     if @rental.save
-      redirect_to @rental, notice: 'Rental was successfully created.'
+      ConfirmationEmailWorker.perform_async(@rental.id)
+      redirect_to @rental, notice: 'Request for confirming the rental has been send to the owner.'
     else
-      redirect_to new_rental_path, alert: @rental.errors.full_messages.join(', ')
+      redirect_to new_rental_path, notice: 'Something went wrong Please, try again.'
+    end
+  end
+
+  def confirm
+    @rental = Rental.find_by(confirmation_token: params[:confirmation_token], id: params[:rental_id])
+
+    if @rental
+      @rental.update(status: :confirmed)
```

Comment:
I guess, you can use autogenerated method here:
`@rental.confirmed!`
This will set status as confirmed and save to the DB.